rfremoterunner/executor_argparser.py

```diff
@@ -0,0 +1,149 @@
+                                 'running individual files or when using resource files. If more than one extension is '
+                                 'needed, separate them with a colon.\r Examples: `--extension robot`, `-F robot:txt`')
+        parser.add_argument('-s', '--suite',
+                            help='Select test suites to run by name. When this option is used with --test, --include or'
+                                 ' --exclude, only test cases in matching suites and also matching other filtering '
+                                 'criteria are selected. Name can be a simple pattern similarly as with --test and it '
+                                 'can contain parent name separated with a dot. You can specify multiple filters by '
+                                 'concatenating with a colon. For example `-s X.Y` selects suite `Y` '
+                                 'only if its parent is `X`. -s X:Y:Z selects X, Y & Z')
+
+        # Arguments passed into robot.run on the remote host:
+        parser.add_argument('-t', '--test',
+                            help='Select test cases to run by name or long name. Name is case insensitive and'
+                                 ' it can also be a simple pattern where `*` matches anything and `?` matches any '
+                                 'char. To specify multiple, concatenate with a colon. Example: -t Foo*:Bar*')
+        parser.add_argument('-i', '--include',
+                            help='Select test cases to run by tag. Similarly as name  with --test, tag is case and '
+                                 'space insensitive and it is possible to use patterns with `*` and `?` as wildcards. '
+                                 'Tags and patterns can also be combined together with `AND`, `OR`, and `NOT` '
+                                 'operators. Examples: --include foo, --include bar*, --include fooANDbar*')
+        parser.add_argument('-e', '--exclude',
+                            help='Select test cases not to run by tag. These tests are not run even if included with '
+                                 '--include. Tags are matched using the rules explained with --include.')
+
+        parser.add_argument('-L', '--loglevel',
+                            help=' Threshold level for logging. Available levels: TRACE, DEBUG, INFO (default), WARN, '
+                                 'NONE (no logging). Use syntax `LOGLEVEL:DEFAULT` to define the default visible log '
+                                 'level in log files. Examples: --loglevel DEBUG --loglevel DEBUG:INFO')
+        return parser
+
+    def get_log_html_output_location(self):
+        """
+        Determine the local output file location of the log.html based on the input arguments
+
+        :return: Path to where the log html file should be saved
+        :rtype: str
+        """
+        return self._resolve_output_path(self.log, 'remote_log.html')

```

Comment:
there's a chance this, and others below, could be referencing fields that don't exist as theyre dynamically allocated in the __init__. But because theyre only allocated if the user inputs it, there's a chance that it could not exist.
Unless ArgParser.__dict__ contains all of the arguments added to it, even if they aren't used? If thats the case ignore this

Reply:
If they're defined as an argument with argparse.ArgumentParser then then they will be present even if they have not been set by the user.